packages/appstore/src/components/total-assets/total-assets.tsx

```diff
@@ -0,0 +1,154 @@
+import classNames from 'classnames';
+
+type TTotalAssets = {
+    category: TAccountCategory;
+    className?: string;
+};
+
+const TotalAssets = ({ category }: TTotalAssets) => {
+    const { client, common } = useStores();
+    const {
+        account_list,
+        accounts,
+        dxtrade_accounts_list,
+        mt5_login_list,
+        obj_total_balance,
+        has_active_real_account,
+    } = client;
+    const { getExchangeRate } = common;
+
+    const [exchanged_rate_cfd_real, setExchangedRateCfdReal] = React.useState(1);
+    const [exchanged_rate_demo, setExchangedRateDemo] = React.useState(1);
+    const [exchanged_rate_cfd_demo, setExchangedRateCfdDemo] = React.useState(1);
+
+    const isDemo = (account: DetailsOfEachMT5Loginid) => account.account_type === 'demo';
+
+    const cfd_real_currency =
+        mt5_login_list.find((mt5_account: DetailsOfEachMT5Loginid) => !isDemo(mt5_account))?.currency ||
+        dxtrade_accounts_list.find((mt5_accounts: DetailsOfEachMT5Loginid) => !isDemo(mt5_accounts))?.currency;
+
+    const cfd_demo_currency =
+        mt5_login_list.find((mt5_account: DetailsOfEachMT5Loginid) => isDemo(mt5_account))?.currency ||
+        dxtrade_accounts_list.find((mt5_account: DetailsOfEachMT5Loginid) => isDemo(mt5_account))?.currency;
+
+    React.useEffect(() => {
+        const getCurrentExchangeRate = (
+            currency: string,
+            setExchangeRate: React.Dispatch<React.SetStateAction<number>>
+        ) => {
+            getExchangeRate(currency, account_total_balance_currency).then((res: number) => {
+                setExchangeRate(res);
+            });
+        };
+        if (cfd_real_currency !== account_total_balance_currency) {
+            getCurrentExchangeRate(cfd_real_currency, setExchangedRateCfdReal);
+        }
+        if (vrtc_currency !== account_total_balance_currency) {
+            getCurrentExchangeRate(vrtc_currency, setExchangedRateDemo);
+        }
+        if (cfd_demo_currency !== account_total_balance_currency) {
+            getCurrentExchangeRate(cfd_demo_currency, setExchangedRateCfdDemo);
+        }
+    }, []);
+
+    const vrtc_loginid = account_list.find((account: { is_virtual: boolean }) => account.is_virtual).loginid;
+    const vrtc_currency = accounts[vrtc_loginid] ? accounts[vrtc_loginid].currency : 'USD';
+    const account_total_balance_currency = obj_total_balance.currency;
+
+    const getTotalBalanceCfd = (mt5_accounts: Mt5LoginList, is_demo: boolean, exchange_rate: number) => {
+        return mt5_accounts
+            .filter((mt5_account: DetailsOfEachMT5Loginid) => (is_demo ? isDemo(mt5_account) : !isDemo(mt5_account)))
+            .reduce(
+                (
+                    total: {
+                        balance: number;
+                    },
+                    mt5_account: DetailsOfEachMT5Loginid
+                ) => {
+                    total.balance += (mt5_account?.balance ?? 1) * exchange_rate;
```

Comment:
```suggestion
                    total.balance += (mt5_account?.balance ?? 1) * exchange_rate;
```
Why 1? Shouldn't we default to 0 if there is no balance?

Reply:
Yes. Thats correct, will change it